.github/actions/install-dashboards/action.yml

```diff
@@ -0,0 +1,83 @@
+name: 'Install Dashboards with Plugin'
+description: 'Installs OpenSearch Dashboard with a Plugin from github, then checkouts the correct dashboards version for the plugin, configures npm/yarn, and bootstraps Dashboards'
+
+inputs:
+  plugin_name:
+    description: 'The the name of the plugin to use, such as security-dashboards-plugin'
+    required: true
+
+outputs:
+  dashboards-directory:
+    description: "The directory where the dashboards has been configured"
+    value: ${{ steps.determine-dashboards-directory.outputs.dashboards-directory }}
+
+  plugin-directory:
+    description: "The directory where the plugin has been configured"
+    value: ${{ steps.determine-plugin-directory.outputs.plugin-directory }}
+
+
+runs:
+  using: "composite"
+  steps:
+    - id: determine-dashboards-directory
+      run: echo "dashboards-directory=OpenSearch-Dashboards"  >> $GITHUB_OUTPUT
+      shell: bash
+
+    - id: determine-plugin-directory
+      run: echo "::set-output name=plugin-directory::./OpenSearch-Dashboards/plugins/${{ inputs.plugin_name }}"
+      # run: echo "plugin-directory=./OpenSearch-Dashboards/plugins/${{ inputs.plugin_name }}" >> $GITHUB_OUTPUT
+      shell: bash
+
+    - uses: actions/checkout@v2
+      with:
+        path: OpenSearch-Dashboards
+        repository: opensearch-project/OpenSearch-Dashboards
+        ref: 'main'
+        fetch-depth: 0
+
+    - run: mkdir -p plugins
+      working-directory: OpenSearch-Dashboards
+      shell: bash
+
+    - uses: actions/checkout@v2
+      with:
+        path: ${{ steps.determine-plugin-directory.outputs.plugin-directory }}
+        # ref: ${{ github.ref }}
+
+    - id: osd-version
+      continue-on-error: true
+      run: |
+        echo "::set-output name=osd-version::$(cat package.json | jq '.opensearchDashboards.version' | cut -c 2-4)"
+        echo "::set-output name=osd-x-version::$(cat package.json | jq '.opensearchDashboards.version' | cut -c 2-3)"
+      working-directory: ${{ steps.determine-plugin-directory.outputs.plugin-directory }}
+      shell: bash
+
+    - id: branch-switch-if-possible
+      continue-on-error: true # Defaults onto main if the branch switch doesn't work
+      if: ${{ steps.osd-version.outputs.osd-version }}
+      run: git checkout ${{ steps.osd-version.outputs.osd-version }} || git checkout ${{ steps.osd-version.outputs.osd-x-version }}x
```

Comment:
Is this case-sensitive?

Reply:
> all branch and tag names are case sensitive by design

_From https://stackoverflow.com/a/57754769/533057_

Sounds like it.